PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostConfigurationViewModel.kt

```diff
@@ -167,6 +187,30 @@ class AzureFunctionsHostConfigurationViewModel(
         handleChangeTfmSelection()
     }
 
+    private fun handleFunctionNameChange(functionName: String) {
+        composeUrlString(port = null, functionName = functionName)
+    }
+
+    private fun composeUrlString(port: Int? = null, functionName: String? = null) {
+        if (port == null && functionName == null)
+            throw IllegalStateException("Missing all parameters (port, functionName). Please provide at least one.")
+
+        val currentUrl = urlEditor.text.value
+        val originalUrl = if (currentUrl.isNotEmpty()) currentUrl else "http://localhost"
+        val url = URIBuilder(originalUrl)
+
+        if (port != null)
+            url.port = port
+
+        if (functionName != null)
+            url.path = if (functionName.isEmpty()) "" else "/api/${functionName.trim()}"
```

Comment:
Right now, we support comma-separated list of function names (e.g. run just these two functions). This situation will break the logic here.

Reply:
Hm. Does it make any sense to patch url if there is only one function name? Or... do you prefer to ignore function name in Url string?

Reply:
Fixed to use a function name only in case a single function is provided (I found it might be helpful for case when we run run gutter mark). Still open for suggestions here :)

Reply:
If we can pick the first Http trigger function would be cool (e.g. for timer trigger this will return a 404)

Reply:
True. I've removed this logic for now. It might be not obvious when you have one Http and one Timer functions. Path is different for them, so, it sounds unclear why do we set http function url over timer trigger. I'll leave it for further enhancements :) I've also found it might be useful to have a text area with feed you can provide to test your http trigger.